### ISSUE TITLE
Odd MapQ result compared to BWA

### ISSUE THREAD

**Issue by kdmurray91** [Cibiv/NextGenMap#35]
Hi all,

First of all, many apologies. I thought I'd posted an issue here on this topic some time ago, but it appears that is not the case.

We have encountered a significant discrepancy between the mapping qualities reported by BWA, and by NGM. Specifically, a reasonable proportion (approx 1/3) of reads have quality scores less than 10 when using NGM in default sensitivity mode.  The actual alignment of a read was identical in most cases, and there was no difference between aligners in the number of alignments for a read (at least on a few screenfuls of alignments, and on average). This caused us to filter out a significant proportion of our data. Note that this happened relatively equally across all ~1500 sequencing runs (of ~500 distinct samples). There are about 200 samples with approx 10x coverage, and the remainder have approx 5x.

 We have a blog post [on my blog](https://www.kdmurray.id.au/posts/2018/02/2018-01_nextgenmappers/) describing this, and the effects of this issue, in further detail.

A couple of questions:

- Does this behaviour sound deliberate on NGM's part, or more like a bug? If this is deliberate, what could we have mis-configured to cause such low mapping qualities?
- If this sounds like a bug, how should we help debug it? I can post logs, however the data is not yet public (though I can send a subset).

Thanks,
Kevin

**Comment by fritzsedlazeck** [comment]
Hi Kevin,
thanks for reaching out. Mapping qualities are always hard to compare directly. We did compare our mapping qualities to Bowtie (1 & 2) and BWA (aln, mem). It appears to me that BWA mem often assigns a MQ of 60 if it can align the second pair in close proximity (independent if that pair is aligned with high confidence). If you compare further mapper you will see they all behave differently. 

For NGM we desired to make it clear and easy. If MQ=0 it means that a equally good alignment can be obtain at a different position. if MQ>0 means that this alignment is unique, but e.g. MQ=1 just one additional substitution can lead to a different position. For example Bowtie says that MQ: 0-3 means a read is mappable to a different position with the same score.

Furthermore, I want to mention that repeats and evolutionary distance will impact the MQ. E.g. if the sample is diverse it may obtain enough edit operations per read such that the region (given those edit distances) is not unique anymore. 

So no its not a bug that we know of, but a behaviour that we know of. 
Here is also a tool to investigate this over some simulated data: 
http://teaser.cibiv.univie.ac.at/

Let me know if that helps and sorry for the confusion. 
Thanks
Fritz

**Comment by kdmurray91** [comment]
Thanks Fritz, this explains a lot!! Sorry we have mis-interpreted the output.

Does any part of the range of mapping qualities correspond to the usual 10*-log10(P[incorrect]) interpretation of mapping qualities? Or does Q=60 mean that additional mappings could be found when mutating 60 bases? And Q=59? It seems 60 could mean "definately a unique mapping" based on the distribution of quality scores we see:

![](https://www.kdmurray.id.au/posts/img/2018-01-30_qualinsert/mapping-quality-barplot.png)

**Comment by fritzsedlazeck** [comment]
This formular was suggested by Heng Li like almost 10 years ago. You will see that every mapper implements MQ slightly different and more or less reassembles it. In all fairness NGM is a bit more pessimistic and thats on purpose. 

So if you have higher mutations rates or higher seq. error rates this will impact the overall MQ for NGM. 
In general everything with MQ>0 is unique for NGM. I usually use MQ>10 for NGM as filter.

So from you plot above I expect some of our MQ1-10 be reported in the range of 20 -50 from BWA mem. Plus then multiple MQ0 (NGM) reads as MQ60 from BWA mem. I think the reason he does that is that he finds pairs in close proximity, but they can also be uncertain. I have invested quite some work in figuring out why or what BWA mem does but did not come to an obvious solution. That is why we implemented the MQ comparison in Teaser to illustrate that every mapper is giving different approximation depending on the information at hand.

@philres can jump in if I say something stupid :)
Cheers
Fritz
